ush/setup.py

```diff
@@ -505,123 +512,62 @@ def get_location(xcs,fmt):
-        print_err_msg_exit(
-            f'''
-            DATE_FIRST_CYCL must be a string consisting of exactly 10 digits of the
-            form \"YYYYMMDDHH\", where YYYY is the 4-digit year, MM is the 2-digit
-            month, DD is the 2-digit day-of-month, and HH is the 2-digit
-            cycle hour.
-              DATE_FIRST_CYCL = \"{DATE_FIRST_CYCL}\"'''
-        )
+        raise Exception(f"\nRESTART_INTERVAL = {RESTART_INTERVAL}, must be an integer value\n")
 
-    if not isinstance(DATE_LAST_CYCL, datetime.date):
-        print_err_msg_exit(
-            f'''
-            DATE_LAST_CYCL must be a string consisting of exactly 10 digits of the
-            form \"YYYYMMDDHH\", where YYYY is the 4-digit year, MM is the 2-digit
-            month, DD is the 2-digit day-of-month, and HH is the 2-digit
-            cycle hour.
-              DATE_LAST_CYCL = \"{DATE_LAST_CYCL}\"'''
-        )
-    #
-    # -----------------------------------------------------------------------
-    #
-    # Call a function to generate the array ALL_CDATES containing the cycle
-    # dates/hours for which to run forecasts.  The elements of this array
-    # will have the form YYYYMMDDHH.  They are the starting dates/times of
-    # the forecasts that will be run in the experiment.  Then set NUM_CYCLES
-    # to the number of elements in this array.
-    #
-    # -----------------------------------------------------------------------
-    #
+    # Check that input dates are in a date format
 
-    ALL_CDATES = set_cycle_dates(
-        date_start=DATE_FIRST_CYCL,
-        date_end=DATE_LAST_CYCL,
-        incr_cycl_freq=INCR_CYCL_FREQ,
-    )
+    # get dictionary of all variables
+    allvars = dict(globals())
+    allvars.update(locals())
+    dates = ['DATE_FIRST_CYCL', 'DATE_LAST_CYCL']
+    for val in dates:
+        if not isinstance(allvars[val], datetime.date):
+            raise Exception(dedent(f'''
+                            Date variable {val}={allvars[val]} is not in a valid date format
 
-    NUM_CYCLES = len(ALL_CDATES)
+                            For examples of valid formats, see the users guide.
+                            '''))
 
-    # Completely arbitrary cutoff of 90 cycles.
-    if NUM_CYCLES > 90:
-        ALL_CDATES = None
-        print_info_msg(
-            f"""
-            Too many cycles in ALL_CDATES to list, redefining in abbreviated form."
-            ALL_CDATES="{DATE_FIRST_CYCL}...{DATE_LAST_CYCL}"""
-        )
-    #
-    # -----------------------------------------------------------------------
-    #
     # If using a custom post configuration file, make sure that it exists.
-    #
-    # -----------------------------------------------------------------------
-    #
     if USE_CUSTOM_POST_CONFIG_FILE:
-        if not os.path.exists(CUSTOM_POST_CONFIG_FP):
-            print_err_msg_exit(
+        try:
+            if not os.path.exists(CUSTOM_POST_CONFIG_FP):
+                raise
```

Comment:
What are you raising here? Why does it need to be in a try block?

Reply:
Because I'd like to check for the existence of this file and error out appropriately here in this script, even though we don't need to take any action on this file until later in the workflow. Do you know of a more elegant way to do this?

Reply:
My first thought is to raise the exception inside the if block if the file doesn't exist. I know you'd mentioned the logger.exception needs to be used in a try block though. You don't seem to be using that here. Just curious.

Reply:
Ah, I see the issue. os.path.exists returns an exception if passed "None", so this structure allows us to catch all errors in one. I added a clarifying comment for future developers.